cmd/config_update.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2025 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/util"
+
+	"github.com/urfave/cli/v3"
+)
+
+func cmdConfig() *cli.Command {
+	subcmdConfigUpdateIni := &cli.Command{
+		Name:  "update-ini",
+		Usage: "Load an existing INI file, apply environment variables, keep specified keys, and output to a new INI file.",
+		Description: `
+Help users to update the gitea configuration INI file:
+* Keep specified keys to for the new INI file.
+* Map environment variables to values in the INI.
+
+# Keep Specified Keys
+
+If you need to re-create the configuration file with only a subset of keys,
+you can provide an INI template file and use the "--config-key-template" flag.
+For example, if a helm chart needs to reset the settings and only keep SECRET_KEY,
+it can use a template file like:
+
+  [security]
+  SECRET_KEY=
+
+$ ./gitea config update-ini --config app-old.ini --config-key-template app-template.ini --out app-new.ini
+
+# Map Environment Variables to INI Configuration
+
+Environment variables of the form "GITEA__section_name__KEY_NAME"
+will be mapped to the ini section "[section_name]" and the key
+"KEY_NAME" with the value as provided.
+
+Environment variables of the form "GITEA__section_name__KEY_NAME__FILE"
+will be mapped to the ini section "[section_name]" and the key
+"KEY_NAME" with the value loaded from the specified file.
+
+Environment variable keys can only contain characters "0-9A-Z_",
+if a section or key name contains dot ".", it needs to be escaped as _0x2E_.
+For example, to apply this config:
+
+	[git.config]
+	foo.bar=val
+
+$ export GITEA__git_0x2E_config__foo_0x2E_bar=val
+
+# Put All Together
+
+$ ./gitea config update-ini --config app.ini --config-key-template app-template.ini --apply-env
+`,
+		Flags: []cli.Flag{
+			// "--config" flag is provided by global flags, and this flag is also used by "environment-to-ini" script wrapper
+			&cli.StringFlag{
+				Name:  "config-key-template",
```

Comment:
This reads wrong to me - template implies the file being templated by something while this is more like a list of keys that shouldn't be touched.

`const-keys` (or similar) would feel slightly more appropriate IMO

Reply:
Hmm, I agree that it is arguable. I did my best to use `key-template` to avoid misunderstanding: a template for "keys"


I don't think `--persist-keys` reads better, it reads like it accept the keys from the CLI but not from a file.

Reply:
By the way, I also have thought about other names like `keep-keys-from` etc. But I decided to use `config-key-template` at last, it is the clearest one in my  mind  at the moment.

The `--config-` prefixes shows that it only works on the `--config` input file, but doesn't affect other keys (at the moment `--apply-env` or in the future `--merge-from=another-config.ini` . `key-template` means it is a template for keys, but not for values.

Reply:
`--config-keep-keys` maybe? That way keep config prefix, it reads more like you're keeping keys from config instead of giving the idea that it's templating them (which is more what the actual command does).